### ISSUE TITLE
[feature] Add parallel ingestion for assembler being used by NATS

### ISSUE THREAD

**Issue by pxp928** [guacsec/guac#963]
**Is your feature request related to a problem? Please describe.**

Currently, the assembler being used by [guacIngest](https://github.com/guacsec/guac/blob/main/cmd/guacingest/cmd/ingest.go#L185) is not running in parallel. [guacone collect](https://github.com/guacsec/guac/blob/main/cmd/guacone/cmd/files.go#L250) is utilizing the [parallel assembler](https://github.com/guacsec/guac/blob/main/pkg/assembler/clients/helpers/parallel.go#L28).

**Describe the solution you'd like**

1. They both need to be consistent and use the parallel assembler to speed up the ingestion of data. 
2. Tests need to be conducted to ensure functionality does not break for [guacIngest](https://github.com/guacsec/guac/blob/main/cmd/guacingest/cmd/ingest.go#L185) by running:
- `make container`
- `docker compose up --force-recreate`
- `go run ./cmd/guaccollect files ../guac-data/docs/spdx/spdx_vuln.json`

Check that there is no errors in the docker logs for ingestion and it returns logs similar to this:

```
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373183.041415,"caller":"parser/parser.go:126","msg":"parsing document tree with root type: SPDX"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373183.0902941,"caller":"helpers/assembler.go:33","msg":"assembling CertifyScorecard: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373183.0903711,"caller":"helpers/assembler.go:40","msg":"assembling IsDependency: 7161"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.0002403,"caller":"helpers/assembler.go:47","msg":"assembling IsOccurence: 3878"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.9143217,"caller":"helpers/assembler.go:54","msg":"assembling HasSLSA: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.9145687,"caller":"helpers/assembler.go:61","msg":"assembling CertifyVuln: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.914619,"caller":"helpers/assembler.go:68","msg":"assembling IsVuln: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.914776,"caller":"helpers/assembler.go:75","msg":"assembling HasSourceAt: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.9148247,"caller":"helpers/assembler.go:82","msg":"assembling CertifyBad: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.914876,"caller":"helpers/assembler.go:89","msg":"assembling CertifyGood: 0"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.9150603,"caller":"helpers/assembler.go:96","msg":"assembling HasSBOM: 1"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.9159975,"caller":"cmd/ingest.go:109","msg":"got collect entries to add: 126"}
artifact-ff-guac-ingestor-1      | {"level":"info","ts":1687373185.916621,"caller":"parser/parser.go:108","msg":"[ingestor: d7251edf-b64a-48e9-a38b-035145950ad0] ingested docTree: {Collector:FileCollector Source:file:///../guac-data/docs/spdx/spdx_vuln.json}"}
```

**Comment by pxp928** [comment]
This issue is on hold under we determine if this feature is needed.

**Comment by lumjjb** [comment]
I think with the batch ingestion with #999 , this should no longer be needed? Since we are leaving concurrency to the backend which may be better at handling it.

**Comment by pxp928** [comment]
yes agreed. Closing issue in favor of batch ingestion.
